src/main/scala/com/guidewire/cda/SavepointsProcessor.scala

```diff
@@ -23,22 +26,26 @@ class SavepointsProcessor(directoryPath: String) {
   private[cda] val savepointsDirPath = directoryPath
   private[cda] val savepointsFileName = "savepoints.json"
   private[cda] val savepointsFilePath = s"${this.savepointsDirPath}/$savepointsFileName"
+  private val savepointsFileURI = new URI(savepointsFilePath)
   private[cda] val savepointsFileExists = checkExists()
   private[cda] val savepointsDataMap: SavepointData = readSavepointsFile()
-
+  
   /** Check to see if there has been a savepoints json that already exists and return true or false.
    * Additionally, validate that the directory under which the savepoints json can be found does exist if
    * no savepoint currently exists, so that we can write to it.
    *
    * @return Boolean if there exists an already written savepoints json
    */
   private[cda] def checkExists(): Boolean = {
-    if (!Files.isDirectory(Paths.get(this.savepointsDirPath))) {
-      throw new IOException(s"Savepoints path ${this.savepointsDirPath} doesn't exist or isn't a directory")
+    savepointsFileURI.getScheme match {
+      case "s3" => AWSUtils.S3Utils.doesFileExist(savepointsFilePath)
+      case _    =>
```

Comment:
I think it would be better to account for all the possibilities by erroring out if the scheme isn't `s3` or `file`.